x-pack/platform/plugins/shared/streams/server/routes/internal/streams/processing/grok_suggestions_handler.ts

```diff
@@ -77,7 +77,11 @@ export const handleProcessingGrokSuggestions = async ({
         ? field.ecs_field.replace('@timestamp', 'custom.timestamp')
         : field.ecs_field;
       return {
-        name: isWiredStream ? getOtelFieldName(name) : name,
+        // if the stream is wired, or if it matches the logs-*.otel-* pattern, use the OTEL field names
+        name:
+          isWiredStream || params.path.name.match(/^logs-.*\.otel-/)
```

Comment:
I would move this check outside the loop. We don't need to run it for each field separately since the result will be the same.